src/reflect/scala/reflect/internal/Flags.scala

```diff
@@ -129,6 +129,7 @@ class ModifierFlags {
   final val JAVA_DEFAULTMETHOD = 1L << 47     // symbol is a java default method
   final val JAVA_ENUM          = 1L << 48     // symbol is a java enum
   final val JAVA_ANNOTATION    = 1L << 49     // symbol is a java annotation
+  final val JAVA_SEALED        = 1L << 50     // symbol is a java sealed
```

Comment:
Is there any difference in behavior between Java sealed and Scala sealed? If not, can we reuse the existing Sealed flag instead of adding another one?

Reply:
+1 to reusing `SEALED`. If there is some logic that only applies to Scala `sealed`, we can implement this with `sym.isSealed && !sym.isJava`.